### PR TITLE
Issue #25

### PR DESCRIPTION
**Pull request by jeestr4d**
Fix for Issue #25

### CODE REVIEW

**Comment by jeestr4d** [comment]
local pytest results:

---

tests/static_route_overlap_check/test_static_route_overlap_check.py::test_logic[icurl_outputs0-4.2(7f)-5.2(4d)-FAIL - OUTAGE WARNING!!] PASSED                                        [ 55%]
tests/static_route_overlap_check/test_static_route_overlap_check.py::test_logic[icurl_outputs1-4.2(7f)-5.2(4d)-PASS] PASSED                                                           [ 55%]
tests/static_route_overlap_check/test_static_route_overlap_check.py::test_logic[icurl_outputs2-5.1(1a)-5.2(4d)-PASS] PASSED                                                           [ 56%]
tests/static_route_overlap_check/test_static_route_overlap_check.py::test_logic[icurl_outputs3-4.2(7f)-5.2(4d)-PASS] PASSED                                                           [ 56%]
==================================================================================== 273 passed in 0.28s ====================================================================================
---

File: aci-preupgrade-validation-script.py [hunk 1]
```diff
@@ -3263,6 +3302,78 @@ def subnet_scope_check(index, total_checks, cversion, **kwargs):
...
     return result
 
+def static_route_overlap_check(index, total_checks, cversion, tversion, **kwargs):
+	title = 'Static Route Subnet and BD Subnet Overlap'
+	result = PASS
+	msg = ''
+	headers = ["Node ID", "Node Name", "Recommended Action"]
+	data = []
+	recommended_action = 'Do not use /32 routes that overlap with a BD subnet range '
+	doc_url = 'https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script'
```
[the remark this answers is not in the document]
Comment (monrog2) [reply aci-preupgrade-validation-script.py:3312]: fixed link

File: aci-preupgrade-validation-script.py [hunk 2]
```diff
@@ -3200,6 +3200,45 @@ def n9k_c93108tc_fx3p_interface_down_check(index, total_checks, tversion, **kwar
     print_result(title, result, msg, headers, data, recommended_action=recommended_action, doc_url=doc_url)
     return result
 
+def static_route_overlap_check(index, total_checks, cversion, tversion, **kwargs):
```
[the remark this answers is not in the document]
Comment (monrog2) [reply aci-preupgrade-validation-script.py:3203]: dup function removed

**Review by monrog2 — COMMENTED** [review]
needs to pull in master

**Comment by monrog2** [comment]
pytest outputs:
[Check  1/1] L3out /32 Static Route and BD Subnet Overlap...                                                      FAIL - OUTAGE WARNING!!
  L3out                 /32 Static Route  BD                BD Subnet
  -----                 ----------------  --                ---------
  uni/tn-AA/out-static  81.81.81.81       uni/tn-AA/BD-bd1  81.81.81.1/24

  Recommended Action: Change /32 static route design or target a fixed version
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#l3out-32-overlap-with-bd-subnet


[Check  1/1] L3out /32 Static Route and BD Subnet Overlap...                                                      FAIL - OUTAGE WARNING!!
  L3out                 /32 Static Route  BD                BD Subnet
  -----                 ----------------  --                ---------
  uni/tn-AA/out-static  81.81.81.81       uni/tn-AA/BD-bd1  81.81.81.1/24

  Recommended Action: Change /32 static route design or target a fixed version
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#l3out-32-overlap-with-bd-subnet


[Check  1/1] L3out /32 Static Route and BD Subnet Overlap...                                                                         PASS
[Check  1/1] L3out /32 Static Route and BD Subnet Overlap...                                                                         PASS
[Check  1/1] L3out /32 Static Route and BD Subnet Overlap... 
```
```

File: aci-preupgrade-validation-script.py [hunk 3]
```diff
@@ -3572,6 +3572,70 @@ def unsupported_fec_configuration_ex_check(index, total_checks, sw_cversion, tve
...
+                    bd_dn = 'uni/tn-' + bd_subnet_re.group("tenant") + '/BD-' + bd_subnet_re.group("bd")
+                    subnet_list = bd_to_subnet.get(bd_dn, [])
+                    subnet_list.append(bd_subnet_re.group("subnet"))
+                    bd_to_subnet[bd_dn] = subnet_list
+
+            for static_route, info in staticR_to_vrf.items():
+                for bd in vrf_to_bd[info['vrf']]:
+                    for subnet in bd_to_subnet[bd]:
+                        if IPAddress.ip_in_subnet(static_route, subnet):
+                            # error_message = 'Overlapping between Static Route ' +  static_route + ' in L3out '+ info['l3out'] + ' and BD Subnet ' + subnet + ' in BD ' + bd
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:3629]: remove comment

**Review by monrog2 — DISMISSED** [review]

**Review by monrog2 — APPROVED** [review]

**Pull request merged**